ghost/admin/app/routes/settings/design/index.js

```diff
@@ -7,9 +7,19 @@ export default class SettingsDesignIndexRoute extends AuthenticatedRoute {
     @service customThemeSettings;
     @service modals;
     @service settings;
+    @service store;
 
     confirmModal = null;
     hasConfirmed = false;
+    themes = this.store.findAll('theme');
```

Comment:
Setting properties to promises like this in singletons (routes, controllers, services) isn't good practice because there's no control over when the class instance will be instantiated and the promise is kicked off.

If we know that themes are loaded in a higher-level route and already in the store (eg, by adding the `findAll` to the list of [promises that the `routes/settings/design.js` model hook waits for](https://github.com/TryGhost/Ghost/blob/main/ghost/admin/app/routes/settings/design.js#L19-L23)) then we can confidently can use a non-promise-returning method like `themes = this.store.peekAll('theme')` here.

Reply:
This makes a lot of sense! I've gone ahead and pushed the changes.